cmd/antrea-agent/options.go

```diff
@@ -135,7 +136,12 @@ func (o *Options) validate(args []string) error {
 	}
 
 	if encapMode.SupportsNoEncap() {
-		if !features.DefaultFeatureGate.Enabled(features.AntreaProxy) {
+		// When use noEnCap traffic mode without AntreaProxy, pod-to-service traffic is handled by iptables/ipvs in
+		// root netns, if the endpoint is not local the DNATed traffic will be output to physical network directly
+		// without going back to OVS for Egress NetworkPolicy enforcement, which breaks basic security functionality.
+		// It is not allowed to use in principle, But can force support NoEncap with TrafficDirectRouting environment
+		// parameter for performance.
+		if !features.DefaultFeatureGate.Enabled(features.AntreaProxy) && !env.GetTrafficDirectRouting() {
```

Comment:
we should still print a warning if the user bypasses the AntreaProxy requirement using the new env variable

Reply:
Agree, warning printing has been added to remind this.